CM7/Core/Src/ssi_encoder.c

```diff
@@ -0,0 +1,57 @@
+#include "ssi_encoder.h"
+
+#include <stdlib.h>
+#include <assert.h>
+#include <math.h>
+
+#include "stm32h7xx.h"
+#include "stm32h7xx_hal_spi.h"
+
+#define ENCODER_MAX_COUNT ((1 << 13) - 1)
+
+ssi_encoder_t *ssi_encoder_init(SPI_HandleTypeDef *spi_handle)
+{
+    // make sure the handle exists
+    if (spi_handle == NULL)
```

Comment:
also maybe assert here? I know asserts are kinda bad practice in embedded by for initialization of critical components I like it